includes/context/RequestContext.php

```diff
@@ -206,8 +206,9 @@ public function getUser() {
 		if ( $this->user === null && $wgUserForceAnon ) {
 			$this->user = new User();
 		}
+		// replace the FromToken user with a FromSession user when in a Piggyback session
+		if ( $this->user === null || PBLoginForm::isPiggyback() ) {
```

Comment:
Based on the same input from @grunny and my own second thoughts (great timing), I have decided to make the core change more general and move the Piggyback-specific logic to Piggyback. :-)